### PR TITLE
Feature/eslint

### PR DESCRIPTION
**Pull request by jason-fox**
Update codebase to use ES6
-  Remove JSHint and jshint overrides
-  Add esLint using standard tamia presets
-  Replace var with let/const
-  Fix or disable eslint errors

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# editorconfig.org
```
Comment (fgalan) [inline .eslintignore]: The reference to jshint should be changed in this section https://github.com/telefonicaid/iotagent-ul/blob/master/docs/usermanual.md#coding-guidelines

(Maybe the same in the other associated PRs)

Comment (jason-fox) [reply .eslintignore]: Fixed - 97d31b3 b15e1b4 95d5b7e

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# editorconfig.org
+root = true
+
+[*]
```
Comment (fgalan) [inline .editorconfig]: Should we force a default style for files not matching any extension? Not sure if it is a good idea to over-force in this case as the potential files can be quite different.

Could be the `[*]` section safely removed?

Comment (jason-fox) [reply .editorconfig]:
```

Comment (jason-fox) [reply .editorconfig]: Suggested fix: e55fdb6

Comment (fgalan) [reply .editorconfig]: > The only controversial one is `indent_style = tab` - maybe should probably be redefined for `Dockerfile` only?

Yes, that's controversial... especially when other file formats in the repo are using `indent_style = space` :). Is there any syntactical requirement that enforce tabs in Dockefiles? Or it's a matter of taste decide between spaces and tabs?

In the second case, I'd suggest to let Dockerfile out of the scope of this file.

> There are a couple of `*.sh` bash scripts in the repo - which are inconsistently spaced - should they be added?

Style & homogeneity for JavaScript source code is great. Style & homogeneity for JSON files could be also great. However, going to far trying to cover little file of the repository could be excessive... I'd suggest by the moment leaving out *.sh files.

Comment (jason-fox) [reply .editorconfig]: So something like this: caaffcb 

- Default to Unix line endings and UTF-8
- agnostic over tabs/spaces. 
- Specific  config for  JS, md, yaml etc for the things that the linter cares about.

Note that editorConfig is just a suggestion. Husky should run the prettifier to re-enforce code style.

Comment (fgalan) [reply .editorconfig]: After the last changes the .eseditorcnofig file looks nice :) Thx!

NTC

File: lib/bindings/AMQPBinding.js [hunk 3]
```diff
@@ -190,8 +190,6 @@ function start(callback) {
                                 numRetried++;
                                 return setTimeout(createConnection, retryTime * 1000);
                             }
-                        } else {
```
Comment (fgalan) [inline lib/bindings/AMQPBinding.js]: CC: @fiqare-secmotic

Curiously this is contrary to one of the rules from fiQare project, see https://github.com/telefonicaid/iotagent-ul/pull/411#discussion_r358409771. We enforce one thing (to remove empty else brarnches) and the opposite (to enforce else branch even when empty) at the same time :)

@jason-fox @fiqare-secmotic what do you think?

Comment (jason-fox) [reply lib/bindings/AMQPBinding.js]: This is an `if...else...` construct not an `if...else if...` construct.  

The `return` within the `else` clause is totally unnecessary.

Comment (fiqare-secmotic) [reply lib/bindings/AMQPBinding.js]: Hi! yes, as @jason-fox says, it is not an  `if...else if` construct

Comment (fgalan) [reply lib/bindings/AMQPBinding.js]: Just to check if I'm understanding it correctly...

The MISRA and CERT rules cited at https://github.com/telefonicaid/iotagent-ul/pull/411#discussion_r364121752, prescribe the inclusion of final "else" in "if... else if..." structures *but not* prescribe the use of "else" in simple "if" structured.

@fiqare-secmotic, could you confirm my understanding is correct?

Comment (fgalan) [reply lib/bindings/AMQPBinding.js]: > @fiqare-secmotic, could you confirm my understanding is correct?

Any news?

**Comment by fgalan** [comment]
Should I understand this PR has moved to PR #415 ? Why? (it's fine, just curiosity...)

The question raised at https://github.com/telefonicaid/iotagent-ul/pull/408#discussion_r364717115 still having no answer. @fiqare-secmotic could you have a look, please? Thanks!

**Comment by fgalan** [comment]
> Should I understand this PR has moved to PR #415 ? Why? (it's fine, just curiosity...)

I just see at PR #415 

> Repeats #408 closed in error

Sorry not checking PR #415 before asking this...

**Comment by jason-fox** [comment]
@fgalan - I was just doing some automated scripted housekeeping and instead of updating `origin:master` to `upstream:master`,  my script accidentally updated `origin:feature/eslint` to `upstream:master` which caused the PR to close.

File: lib/bindings/AMQPBinding.js (hunk 3 above)
Comment (fiqare-secmotic) [reply lib/bindings/AMQPBinding.js]: @fgalan yes, your understanding is correct.

**Pull request closed without merging**